library/src/main/java/bio/terra/landingzone/stairway/flight/create/resource/step/BaseResourceCreateStep.java

```diff
@@ -179,6 +180,7 @@ private StepResult handleManagementException(
         landingZoneId,
         ManagementExceptionUtils.buildErrorInfo(e));
     MetricUtils.incrementLandingZoneCreationFailure(landingZoneType);
-    return new StepResult(StepStatus.STEP_RESULT_FAILURE_FATAL, e);
+    return new StepResult(
```

Comment:
The description says you extended error handling, but actually `FlightExceptionTranslator` already existed and handled this specific exception, correct?

```
Extend error handling so this can be used in different scenarios (using existing FlightExceptionTranslator with corresponding rules). Currently only one rule exist - BatchAccountQuotaExceedExceptionRule

Reply:
Yes. This mechanism already exists. By extend I mean we can handle properly quota issue.

Reply:
But the problem is that it wasn't used in stairway-based LZ creation. It was used in the original LZ creation implementation here - https://github.com/DataBiosphere/terra-landing-zone-service/blob/608d7b7d88daea05577baf3eb7ed428dbfc0574a/library/src/main/java/bio/terra/landingzone/stairway/flight/create/CreateAzureLandingZoneStep.java#L82.

Reply:
Now if we see similar issue (not necessarily quota, but something not user-friendly) we need to introduce specific rule and register it in FlightExceptionTranslator.

Reply:
I updated PR description as well.